Flutter - Adds Portuguese translation for layout folder (#867)
ZupIT/beagle-docs: content/pt/flutter/layout/styles.md

}
```

> Por padrão, componentes com estilo desabilitado expandem-se para preencher todo o espaço disponível. Para desabilitar este comportamento, você pode alterar a propriedade `shoudExpand` para falso. Utilize está funcionalidade com cuidado pois ela pode gerar erros de layout no Flutter.

Reviewer: ```suggestion
Por padrão, componentes sem um estilo declarado expandem-se para preencher todo o espaço disponível. Para desabilitar este comportamento, você pode alterar a propriedade `shoudExpand` para falso. Utilize está funcionalidade com cuidado pois ela pode gerar erros de layout no Flutter.
```

}
```

# Controlando "padding" e "decoration"

> Alguns widgets precisam eles mesmos tratarem seus estilos, por exemplo, adicionar "wrappers" não funciona. Veja o caso do botão, como usamos um widget do tipo `ElevatedButton`, adicionar "background color", "borders" e "padding" ao "wrapper" não vai funcionar. Ao invés disso, um objecto do tipo `ButtonStyle` deve ser criado baseado no estilo desejado e então passado ao `ElevatedButton`. Precisamos também desabilitar o comportamento padrão para padding e decorations ("backgroundcolor" e "border")

Reviewer: ```suggestion
Alguns widgets tratam seus próprios estilos, por exemplo, adicionar "wrappers" não funciona. Veja o caso do botão, como usamos um widget do tipo `ElevatedButton`, adicionar "background color", "borders" e "padding" ao "wrapper" não vai funcionar. Ao invés disso, um objecto do tipo `ButtonStyle` deve ser criado baseado no estilo desejado e então passado ao `ElevatedButton`. Precisamos também desabilitar o comportamento padrão para padding e decorations ("backgroundcolor" e "border")
```

Para desabilitar o padding e decoration automáticos, mas manter todos os outros estilos, podemos implementar um método `getStyleConfig` do `ComponentBuilder` e retornar a configuração onde `` tem o valor de "true" mas `shouldDecorate` tem o valor de "false"

> Basta então passá-lo para o widget através do `ComponentBuilder`. Veja no trecho de código abaixo:

Reviewer: ```suggestion
Basta então passá-lo para o widget através do `ComponentBuilder`. Veja como no trecho de código abaixo:
```

}}
```

`style` é do tipo `BeagleStyle`.

# Criando seus próprios containers

> Através do pacto beagle_components, a biblioteca fornce um componente muito importante chamado `Container`, O container é responsável por manter e organizar multiplos componentes de acordo com as propriedades providas do flex (`style.flex`).

Reviewer: ```suggestion
Através do pacto beagle_components, a biblioteca fornece um componente muito importante chamado `Container`, O container é responsável por manter e organizar multiplos componentes de acordo com as propriedades do flex (`style.flex`) providas.
```

> Através do pacto beagle_components, a biblioteca fornce um componente muito importante chamado `Container`, O container é responsável por manter e organizar multiplos componentes de acordo com as propriedades providas do flex (`style.flex`).

Reviewer: Do you think "pacto" is the best word here? Shouldn't we use "contract" or something in this way? I know pact and contact are close, but is it usual in Flutter? I honestly don't know. :-) Feel free to ignore this.

Uma linguagem de estilo que funciona para todas as plataformas foi muito importante para iOS, Android e Web. Assim, usando o Yoga, implementamos um conjunto de recursos de layout com base no layout Flex da especificação CSS. Agora, se estamos usando o Flutter, ainda precisamos usar a mesma linguagem de estilo? Ninguém vai desenvolver um app para iOS e Flutter, por exemplo, seria uma loucura. Como todas as plataformas do aplicativo podem usar o Flutter, a linguagem de estilo poderia ser definida pelo próprio Flutter e não pelo Beagle, o que seria mais simples para desenvolvedores já familiarizados com os widgets do Flutter.

O Beagle Flutter é fornecido com o mecanismo de layout usado por todas as bibliotecas de interface do usuário do Beagle, portanto, ao usar nossos componentes padrão, a propriedade "style" pode ser usada e funciona. Dessa forma, é mais fácil reproduzir os exemplos, pois eles não são específicos da plataforma.

> Embora os estilos Beagle funcionem na maioria das vezes, infelizmente [eles têm alguns problemas](https://gist.github.com/Tiagoperes/12f786e06ad80752afacaf80bc49bcf9). Além disso, se você for usar apenas o Flutter para desenvolver o aplicativo e a plataforma não for realmente um problema, pode ser mais fácil se adaptar a um mecanismo de layout no backend que funcione mais como o próprio Flutter.

Reviewer: ```suggestion
Embora os estilos Beagle funcionem na maioria das vezes, infelizmente, [alguns ajustes ainda são necessários](https://gist.github.com/Tiagoperes/12f786e06ad80752afacaf80bc49bcf9). Além disso, se você for usar apenas o Flutter para desenvolver o aplicativo e a plataforma não for realmente um problema, pode ser mais fácil se adaptar a um mecanismo de layout no backend que funcione mais como o próprio Flutter.
```

}
```

> Acima está um exemplo de como poderíamos ignorar o motor padrão de layout do Beagle e usar o nosso customizado, apenas expondo alguns componentes de layout do Flutter.

Reviewer: ```suggestion
Acima temos um exemplo de como poderíamos ignorar o motor padrão de layout do Beagle e usar o nosso customizado, apenas expondo alguns componentes de layout do Flutter.
```
